src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -460,14 +460,17 @@ public function setCaseSensitiveEncoding($encoding)
      */
     private function getCaseSensitiveEncoding()
     {
-        if (!$this->caseSensitiveEncoding) {
-            $params = $this->conn->getParams();
-            $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
+        if ($this->caseSensitiveEncoding) {
+            return $this->caseSensitiveEncoding;
+        }
 
+        $params = $this->conn->getParams();
+        $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
+        if (!isset($params['defaultTableOptions']) || !isset($params['defaultTableOptions']['collate'])) {
             return $this->caseSensitiveEncoding = $charset === 'binary' ? $charset : $charset . '_bin';
         }
 
-        return $this->caseSensitiveEncoding;
+        return $this->caseSensitiveEncoding = $params['defaultTableOptions']['collate'];
```

Comment:
i'd prefer to invert this: `if (isset && isset) { return ... 'collate'; }` and then the fallback to charset binary or not.